lib/new_relixir/current_transaction.ex

```diff
@@ -0,0 +1,44 @@
+defmodule NewRelixir.CurrentTransaction do
+  @doc """
+  Keeps track of the New Relic transaction name for the current process.
+
+  As soon as a request starts and its transaction name is determined, that
+  name is stored in the process dictionary with `CurrentTransaction.get`.
+
+  Later on, spawned descendent processes can still record custom calls and
+  database operations under the same parent New Relic transaction.
+  """
+  @key :new_relixir_transaction
+
+  def get do
+    Process.get(@key) || search_on_ancestors()
```

Comment:
If we return `nil` here, it may fail somewhere when recording in NewRelic.
I don't if the `record_xxx` functions are expecting `nil` values, would be better fail here with an `exception`?
Or maybe prevent to report to NewRelic when we have a `nil` transaction and print a warning?

Reply:
@ulissesalmeida Good point, maybe we should make it an `{:ok, transaction} | {:error, :no_transaction}` function 👍